File: mandelbrot/mandelbrot.lua
-- The Computer Language Shootout
-- http://shootout.alioth.debian.org/
-- contributed by Mike Pall

local function run(n)
  start_time = os.clock()

  local width = n
  local height, wscale = width, 2/width
  local m, limit2 = 50, 4.0
  local write, char = io.write, string.char

  write("P4\n", width, " ", height, "\n")

  for y=0,height-1 do
    local Ci = 2*y / height - 1
    for xb=0,width-1,8 do
      local bits = 0
      local xbb = xb+7
      for x=xb,xbb < width and xbb or width-1 do
        bits = bits + bits
        local Zr, Zi, Zrq, Ziq = 0.0, 0.0, 0.0, 0.0
        local Cr = x * wscale - 1.5
        for i=1,m do
          local Zri = Zr*Zi
          Zr = Zrq - Ziq + Cr
          Zi = Zri + Zri + Ci
          Zrq = Zr*Zr
          Ziq = Zi*Zi
          if Zrq + Ziq > limit2 then
            bits = bits + 1
            break
          end
        end
      end
      if xbb >= width then
        for x=width,xbb do bits = bits + bits + 1 end
      end
      write(char(255-bits))
    end
  end

  io.stderr:write(string.format("time(%.9f)\n", os.clock() - start_time))
end

local N = tonumber(arg and arg[1]) or 100
local times = tonumber(arg and arg[2]) or 1

io.stderr:write("started")

for i=1,times,1 do
  run(N)
end

File: mandelbrot/mandelbrot.php
<?php
/* The Computer Language Benchmarks Game
   https://salsa.debian.org/benchmarksgame-team/benchmarksgame/
   contributed by Thomas GODART (based on Greg Buchholz's C program) 
   modified by anon
*/

function run($n) { 
  $start = microtime(true);
  ob_implicit_flush(1);
  ob_start(NULL, 4096);


  $h = $n;
  $w = $h;

  printf ("P4\n%d %d\n", $w, $h);

  $bit_num = 128;
  $byte_acc = 0;

  $yfac = 2.0 / $h;
  $xfac = 2.0 / $w;

  for ($y = 0 ; $y < $h ; ++$y)
  {
     $result = array('c*');

     $Ci = $y * $yfac - 1.0;

     for ($x = 0 ; $x < $w ; ++$x)
     {
        $Zr = 0; $Zi = 0; $Tr = 0; $Ti = 0.0;

        $Cr = $x * $xfac - 1.5;

        do {
           for ($i = 0 ; $i < 50 ; ++$i)
           {
              $Zi = 2.0 * $Zr * $Zi + $Ci;
              $Zr = $Tr - $Ti + $Cr;
              $Tr = $Zr * $Zr;
              if (($Tr+($Ti = $Zi * $Zi)) > 4.0) break 2;
           }
           $byte_acc += $bit_num;
        } while (FALSE);

        if ($bit_num === 1) {
           $result[] = $byte_acc;
           $bit_num = 128;
           $byte_acc = 0;
        } else {
           $bit_num >>= 1;
        }
     }
     if ($bit_num !== 128) {
        $result[] = $byte_acc;
        $bit_num = 128;
        $byte_acc = 0;
     }
     echo call_user_func_array('pack', $result);
  }
  fprintf(STDERR, "time(%.9f)\n", microtime(true) - $start);
}

$n = ($argc >= 2) ? $argv[1] : 100;
$times = ($argc == 3) ? $argv[2] : 1;

fprintf(STDERR, "started\n");

for($i = 1; $i <= $times; ++$i) {
  run($n);
}


?>